### PR TITLE
Adding new kernel manager functionality and unit tests 

### PR DESCRIPTION
**Pull request by esimpsons3ti**
Kernel manager now creates temporary directory regardless of whether or not environment variable `CSPYCE_TEST_KERNELS` is set. Three tests in test_d.py are still currently failing and will be addressed in new or existing issues.

### CODE REVIEW

File: unittests/gettestkernels.py [hunk 1]
```diff
@@ -22,38 +22,62 @@
...
+    except KeyError:
+        pass
+
+    directory = Path(tempfile.gettempdir()) / "cspyce-test-kernels"
+    directory.mkdir(parents=True, exist_ok=True)
+    return str(directory)
+
+
+def _get_test_output_directory():
+    directory = Path(tempfile.gettempdir()) / "test_file_output"
```
Comment (rfrenchseti) [inline unittests/gettestkernels.py:49]: Let's be consistent about dashes (cspyce-test-kernels) vs underscores (test_file_output). Dashes seem to be more common in filenames. Also let's put "cspyce" in the test-file-output name just so it doesn't conflict with any other program.

File: unittests/gettestkernels.py [hunk 2]
```diff
@@ -22,38 +22,62 @@
...
     return url.split("/")[-1]
 
 
 def get_path_from_url(url: str) -> str:
-    return os.path.join(cwd, get_kernel_name_from_url(url))
+    return os.path.join(KERNEL_DIR, get_kernel_name_from_url(url))
 
 
-def cleanup_file(path: str) -> None:
+def cleanup_file(_path: str) -> None:
```
Comment (rfrenchseti) [inline unittests/gettestkernels.py:65]: Not sure why this has an underscore now.

File: unittests/gettestkernels.py [hunk 3]
```diff
@@ -22,38 +22,62 @@
...
 # =============================================================================
 #     if os.path.exists(path):
 #         os.remove(path)
 # =============================================================================
     pass
 
-class CassiniKernels(object):
+
+def delete_cache_directory():
```
Comment (rfrenchseti) [inline unittests/gettestkernels.py:73]: This makes me really nervous to have here. I don't think we should ever have a function that deletes an entire directory tree based on the contents of an environment variable. That could lead to all sorts of malicious behavior.

**Review by rfrenchseti — CHANGES_REQUESTED** [review]
Some of these comments may be on changes Frank made

File: unittests/gettestkernels.py (hunk 3 above)
Comment (esimpsons3ti) [reply unittests/gettestkernels.py:73]: This is from Frank's work on the temporary directories. Should this function be deleted or commented out? It does state that it's only to be used in the event of kernel corruption.

Comment (rfrenchseti) [reply unittests/gettestkernels.py:73]: Just delete it. If you need to delete a directory, just do it from the command line.

**Review by rfrenchseti — APPROVED** [review]

File: unittests/gettestkernels.py (hunk 3 above)
Comment (fyellin) [reply unittests/gettestkernels.py]: The reason I added this was that at least on my machine, the directory is off someplace strange.  It's not obvious what to delete.  In my case, one of the download files got corrupted (I was watching the download from my debugger and interrupted it before the download was completed.). So I had a corrupt file, and no way to say if its corrupt, and no way to figure out where it was.

Comment (rfrenchseti) [reply unittests/gettestkernels.py]: Maybe you can just add a separate program to print the directory name?

**Pull request merged**